beater/config.go

```diff
@@ -116,17 +117,26 @@ func (s *SourceMapping) isSetup() bool {
 	return s != nil && (s.EsConfig != nil)
 }
 
-func (c *FrontendConfig) memoizedSmapMapper() (sourcemap.Mapper, error) {
-	smap := c.SourceMapping
-	if !c.isEnabled() || !smap.isSetup() {
+func (c *Config) SetRumConfig() {
+	if c.RumConfig != nil && c.RumConfig.Enabled != nil {
+		return
+	}
+	if c.FrontendConfig != nil && c.FrontendConfig.Enabled != nil {
+		c.RumConfig = c.FrontendConfig
+	}
+}
+
+func (c *rumConfig) memoizedSmapMapper() (sourcemap.Mapper, error) {
+	if c == nil || !c.isEnabled() || !c.SourceMapping.isSetup() {
```

Comment:
`c.IsEnabled` already does nil check on `c`, so i think  `c == nil` here shouldn't be needed